src/test/rpc/Peers_test.cpp

```diff
@@ -0,0 +1,99 @@
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/Env.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/overlay/Cluster.h>
+#include <ripple/overlay/Overlay.h>
+#include <unordered_map>
+
+namespace ripple {
+
+class Peers_test : public beast::unit_test::suite
+{
+    void testRequest()
+    {
+        testcase("Basic request");
+        using namespace test::jtx;
+        Env env {*this};
+
+        // without modification of the cluster, expect an empty set
+        // from this request
+        auto peers = env.rpc("peers")[jss::result];
+        BEAST_EXPECT(peers.isMember(jss::cluster) &&
+            peers[jss::cluster].size() == 0 );
+        BEAST_EXPECT(peers.isMember(jss::peers) &&
+            peers[jss::peers].isNull());
+
+        // insert some nodes in to the cluster
+        std::unordered_map<std::string, std::string> nodes;
+        for(auto i =0; i < 3; ++i)
+        {
+
+            auto seed = generateSeed("seed" + std::to_string(i));
+            auto pubkey = derivePublicKey(
+                    KeyType::secp256k1,
+                    generateSecretKey(KeyType::secp256k1, seed));
+
+            std::string name = "Node " + std::to_string(i);
+
+            env.app().cluster().update(
+                pubkey,
+                name,
+                200,
+                env.timeKeeper().now() - 10s);
+            nodes.insert( std::make_pair(
+                toBase58(TokenType::TOKEN_NODE_PUBLIC, pubkey), name));
+        }
+
+        // make request, verify nodes we created match
+        // what is reported
+        peers = env.rpc("peers")[jss::result];
+        if(! BEAST_EXPECT(peers.isMember(jss::cluster)))
+            return;
+        if(! BEAST_EXPECT(peers[jss::cluster].size() == nodes.size()))
+            return;
+        for(auto it  = peers[jss::cluster].begin();
+                 it != peers[jss::cluster].end();
+                 ++it)
+        {
+            auto key = it.key().asString();
+            auto search = nodes.find(it.key().asString());
+            if(! BEAST_EXPECTS(search != nodes.end(), it.key().asString()))
+                continue;
+            auto tag = (*it)[jss::tag].asString();
+            BEAST_EXPECTS(
+                (*it)[jss::tag].asString() == nodes[it.key().asString()],
+                it.key().asString());
+        }
+        BEAST_EXPECT(peers.isMember(jss::peers) &&
```

Comment:
I'm guessing `jtx` doesn't have a nice way to add peers in a way to exercise this part of the call?